tdiag/src/commands/arrangements.rs

```diff
@@ -0,0 +1,182 @@
+            //                             None => {
+            //                                 assert!(!map.contains_key(&key));
+            //                                 map.insert(key, ts);
+            //                             }
+            //                             Some(_) => {
+            //                                 assert!(map.contains_key(&key));
+            //                                 let end = map.remove(&key).unwrap();
+            //                                 let ts_clip =
+            //                                     std::time::Duration::from_secs(ts.as_secs() + 1);
+            //                                 let elapsed = ts - end;
+            //                                 let elapsed_ns = (elapsed.as_secs() as isize)
+            //                                     * 1_000_000_000
+            //                                     + (elapsed.subsec_nanos() as isize);
+            //                                 session.give((key.1, ts_clip, elapsed_ns));
+            //                             }
+            //                         }
+            //                     }
+            //                 });
+            //             }
+            //         },
+            //     )
+            //     .as_collection()
+            //     .consolidate()
+            //     .inspect(|x| println!("time {:?}", x));
+
+            // Track sizes.
+            events
+                .flat_map(|(t, worker, x)| match x {
+                    Batch(x) => Some(((worker, x.operator), t, x.length as isize)),
+                    Merge(x) => match x.complete {
+                        None => None,
+                        Some(complete_size) => {
+                            let size_diff =
+                                (complete_size as isize) - (x.length1 + x.length2) as isize;
+
+                            Some(((worker, x.operator), t, size_diff as isize))
+                        }
+                    },
+                    MergeShortfall(x) => {
+                        eprintln!("MergeShortfall {:?}", x);
+                        None
+                    }
+                })
+                .as_collection()
+                .inner
+                // We do not bother with retractions here, because the
+                // user is only interested in the current count.
+                .filter(|(_, _, count)| count >= &0)
+                .as_collection()
+                .delay(move |t| {
+                    let w_secs = window_size.as_secs();
+
+                    let secs_coarsened = if w_secs == 0 {
+                        t.as_secs()
+                    } else {
+                        (t.as_secs() / w_secs + 1) * w_secs
+                    };
+
+                    Duration::new(secs_coarsened, 0)
+                })
+                .count()
+                .join(&operates)
+                .inspect(|x| println!("{:?}", x));
```

Comment:
Is the output unformatted because you're reading it mechanically?

Reply:
No, but I didn't have a preference on formatting either. Happy to change it to something gnuplottable.